data/bookkeeping/block.go

```diff
@@ -516,6 +522,59 @@ func MakeBlock(prev BlockHeader) Block {
 	return blk
 }
 
+// MakeDevModeBlock constructs a new block for dev mode.
+// If a valid DevModeConfigs is passed in, we also populate block header
+// information, such as block seed and timestamp offsets.
+func MakeDevModeBlock(prev BlockHeader, configs DevModeOpts) Block {
```

Comment:
Can `MakeDevModeBlock` possibly call out to `MakeBlock` and then apply any DevMode specific changes? This may be slightly less performant than the current approach, but -if sensible- it could keep the code more in sync for future changes. Also, it would make it clear exactly what the differences are.

Reply:
Great point about keeping it in sync for future changes

refactored to use `MakeBlock` in [d870d7a](https://github.com/algorand/go-algorand/pull/5296/commits/d870d7ab0c3349bdc2e368965a7936751e199f60)